### PR TITLE
Test stake 

### LINKED ISSUE #195: Tests for stake command

### PR DESCRIPTION
**Pull request by Yashk767**
closes #195

### CODE REVIEW

**Comment by Yashk767** [comment]
I have not included the flag inputs and related util methods which are required to derive txnArgs in stake function as we need txnArgs in approve() function as well.
e.g functions like `utils.AssignPassword(cmd.Flags())`, `utils.ConnectToClient(config.Provider)`,   `balance, err := utils.FetchBalance(client, address)` ). So those functions will be tested in some other command function (eg. In delegate.go, we only have one function i.e delegate so we can include those functions in delegate  itself and get them tested there.)

**Comment by ashish10677** [comment]
> I have not included the flag inputs and related util methods which are required to derive txnArgs in stake function as we need txnArgs in approve() function as well.
> e.g functions like `utils.AssignPassword(cmd.Flags())`, `utils.ConnectToClient(config.Provider)`, `balance, err := utils.FetchBalance(client, address)` ). So those functions will be tested in some other command function (eg. In delegate.go, we only have one function i.e delegate so we can include those functions in delegate itself and get them tested there.)

It's okay that they are not tested here, but since these are `utils` functions, shouldn't they be tested in the `utils` package?

File: cmd/interface.go [hunk 1]
```diff
@@ -25,3 +26,7 @@ type tokenManagerInterface interface {
 type transactionInterface interface {
 	Hash(*Types.Transaction) common.Hash
 }
+
+type stakeManagerInterface interface {
+	Stake(*bind.TransactOpts, uint32, *big.Int, *ethclient.Client) (*Types.Transaction, error)
```
Comment (ashish10677) [inline cmd/interface.go:31]: Can you keep *ethclient.Client as the first parameter?

**Review by ashish10677 — CHANGES_REQUESTED** [review]

**Comment by Yashk767** [comment]
> > I have not included the flag inputs and related util methods which are required to derive txnArgs in stake function as we need txnArgs in approve() function as well.
> > e.g functions like `utils.AssignPassword(cmd.Flags())`, `utils.ConnectToClient(config.Provider)`, `balance, err := utils.FetchBalance(client, address)` ). So those functions will be tested in some other command function (eg. In delegate.go, we only have one function i.e delegate so we can include those functions in delegate itself and get them tested there.)
> 
> It's okay that they are not tested here, but since these are `utils` functions, shouldn't they be tested in the `utils` package?

True, I was saying it on a coverage perspective. As we could not include those functions in stake than will the coverage here be fulfilled I am not sure , if not than how can I do it ?

**Comment by ashish10677** [comment]
> > > I have not included the flag inputs and related util methods which are required to derive txnArgs in stake function as we need txnArgs in approve() function as well.
> > > e.g functions like `utils.AssignPassword(cmd.Flags())`, `utils.ConnectToClient(config.Provider)`, `balance, err := utils.FetchBalance(client, address)` ). So those functions will be tested in some other command function (eg. In delegate.go, we only have one function i.e delegate so we can include those functions in delegate itself and get them tested there.)
> > 
> > 
> > It's okay that they are not tested here, but since these are `utils` functions, shouldn't they be tested in the `utils` package?
> 
> True, I was saying it on a coverage perspective. As we could not include those functions in stake than will the coverage here be fulfilled I am not sure , if not than how can I do it ?

It should be fulfilled I guess. Let's see once coveralls is integrated. For the time being it is fine.

**Review by ashish10677 — APPROVED** [review]
LGTM

**Comment by ashish10677** [comment]
Leaving for @SkandaBhat to merge. Looks good to me.

**Comment by ashish10677** [comment]
@Yashk767 please resolve the conflicts.

**Review by SkandaBhat — APPROVED** [review]

**Comment by SkandaBhat** [comment]
@Yashk767 what is the test coverage percentage?

**Comment by Yashk767** [comment]
> @Yashk767 what is the test coverage percentage?

Coverage is not implemented right ?
Is there any other way where I can get it

**Comment by SkandaBhat** [comment]
Run locally and check

**Comment by Yashk767** [comment]
> @Yashk767 what is the test coverage percentage?

Its 100% for stakeCoins function.

**Comment by SkandaBhat** [comment]
What is the overall percentage?


> > @Yashk767 what is the test coverage percentage?
> 
> Its 100% for stakeCoins function.

**Comment by Yashk767** [comment]
> What is the overall percentage?
> 
> > > @Yashk767 what is the test coverage percentage?
> > 
> > 
> > Its 100% for stakeCoins function.

coverage: 20.5% of statements
Overall

**Pull request merged**